PopPUNK/refine.py

```diff
@@ -102,6 +120,15 @@ def refineFit(distMat, sample_names, start_s, mean0, mean1,
     # Optimize boundary - grid search for global minimum
     sys.stderr.write("Trying to optimise score globally\n")
 
+    # Generate sample combinations
+    potential_edges = list(listDistInts(sample_names, sample_names, True))
```

Comment:
Won't this be absolutely huge? Why is a dataframe with all by all edges needed? I really think that should be avoided

Reply:
See explanation above.